dipy/core/gradients.py

```diff
@@ -210,7 +210,25 @@ def bvecs(self):
         denom = self.bvals + (self.bvals == 0)
         denom = denom.reshape((-1, 1))
         return self.gradients / denom
-
+    
+    def __getitem__(self, idx, threshold = 200):
```

Comment:
I would set the threshold to be 0 per default. 

One more thought is to implement a lower and upper bound, so that if you index: 

```
gtab[:, 100:1000]
```

you would get a new gradient table with all the b-values between 100 and 1,000. I can see some use-cases for this kind of selection.

Reply:
I was thinking that we can probably remove the threshold argument and associated code, as it is no longer needed with the new indexing scheme.
Since now we could have: def __getitem__(self, bmin=0, bmax=np.inf):
And to create mask we can simply use: mask = (self.bvals >= bmin) & (self.bvals <= bmax). Using this to create mask we can remove a lot of unnecessary code as well. 
and # Select b-values between 100 and 1000, 
gtab = gtab[bmin=100, bmax=1000]
What do you think of this?

Reply:
I like the idea! 

however, a deprecation cycle will be needed. a lot people are using this default behavior so we will have to keep both behavior for 1 year (2-3 releases) at least.

Reply:
So are you sugggesting something like this? 
def __getitem__(self, bmin=0, bmax=np.inf, threshold=0):
    # Use the new behavior if bmin and bmax are provided
    if bmin != 0 or bmax != np.inf:
        mask = (self.bvals >= bmin) & (self.bvals <= bmax)
    else:
        # Use the old behavior if bmin and bmax are not provided
        mask = self.bvals > threshold

Reply:
> But I don't quite understand the part where the interp parameter is being deprecated from version 1.19 onwards. Can you explain more on that.

typo, replaced by `threshold`

Reply:
Oh right, so should i create a commit for this? I mean the method looks correct.

Reply:
yes, update the code and test it to make sure it works as expected.

Reply:
I modified the gradients file and tests_gradients.py as well. Please review it.

Reply:
Could somebody please manually trigger the checks on my last commit?